src/main/java/com/mixpanel/android/mpmetrics/MixpanelAPI.java

```diff
@@ -193,11 +193,19 @@
      * Use MixpanelAPI.getInstance to get an instance.
      */
     MixpanelAPI(Context context, Future<SharedPreferences> referrerPreferences, String token) {
+        this(context, referrerPreferences, token, MPConfig.getInstance(context));
+    }
+
+    /**
+     * You shouldn't instantiate MixpanelAPI objects directly.
+     * Use MixpanelAPI.getInstance to get an instance.
+     */
+    MixpanelAPI(Context context, Future<SharedPreferences> referrerPreferences, String token, final MPConfig config) {
```

Comment:
Hmmm.. I sort of like this change, but, on the other hand, I don't think it's necessary. As the comment says, the user shouldn't use this constructor to get an instance. The only place uses this is the `getInstance` API. And because of that, I think it's an overkill to provide an extra level of extraction in order to make it more flexible. Do you have a good reason for adding this that I'm not aware of?